src/LinAlg/hiopVectorRajaImpl.hpp

```diff
@@ -413,20 +403,21 @@ void hiopVectorRajaPar::copy_from_indexes(const double* vv, const hiopVectorInt&
  * @pre Size of `v` must be >= nv.
  * @pre start_index_in_this+nv <= n_local_
  * @pre `this` is not distributed
- * 
+ * @pre `v` should be allocated in the memory space/backend of `this`
+ *
  * @warning Method casts away const from the `local_array`.
  */
-void hiopVectorRajaPar::copyFromStarting(int start_index_in_this, const double* v, int nv)
+template<class MEM, class POL>
+void hiopVectorRaja<MEM,POL>::copyFromStarting(int start_index_in_this, const double* v, int nv)
 {
   assert(start_index_in_this+nv <= n_local_);
 
   // If nothing to copy, return.  
   if(nv == 0)
     return;
 
-  auto& rm = umpire::ResourceManager::getInstance();
-  double* vv = const_cast<double*>(v); // <- cast away const
-  rm.copy(data_dev_ + start_index_in_this, vv, nv*sizeof(double));
+  //TODO: data_dev_+start_index_in_this   -> is not portable, may not work on the device. RAJA loop should be used
+  exec_space_.copy(data_dev_+start_index_in_this, v, nv);
```

Comment:
This function hasn't been an issue before AFAIK - does the new implementation introduce this bug? Is there some pre/post conditions that we need to check that would be helpful?

Reply:
it's just that I find it unsafe to do device pointer arithmetic on host. Works fine on both amd and cuda right now.